### PR TITLE
kubecontrollersconfig resource definition

### PR DESCRIPTION
**Pull request by spikecurtis**
Signed-off-by: Spike Curtis <spike@tigera.io>

## Description

Work-in-progress docs for KubeControllersConfiguration API resource.

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
Documentation for KubeControllersConfiguration API resource
```

### CODE REVIEW

**Comment by caseydavenport** [comment]
Deploy preview for *calico* ready!

Built with commit bce6cd40bf91757025f73d509b462ba7584bb13e

https://deploy-preview-3385--calico.netlify.com

File: reference/resources/kubecontrollersconfig.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+The node controller automatically cleans up configuration for nodes that no longer exist. Optionally, it can create host endpoints for all Kubernetes nodes.
+
+| Field                              | Description                 | Accepted Values   | Schema | Default    |
+|------------------------------------|-----------------------------|-------------------|--------|------------|
+| reconcilerPeriod | Period to perform reconciliation with the {{site.prodname}} datastore | | [Duration string][parse-duration] | 5m |
+| numberOfWorkers | Number of goroutines to allocate to the controller | > 0 | int | 1 |
+| syncLabels | When enabled, Kubernetes node labels will be copied to {{site.prodname}} node objects. | Enabled, Disabled | string | Enabled |
+| hostEndpoint | Controls allocation of host endpoints | | [HostEndpoints](#hostendpoints) | |
+
+#### HostEndpoints
```
Comment (lmm) [inline reference/resources/kubecontrollersconfig.md]: Maybe `HostEndpoint`? It doesn't read as nice but could be a point of confusion (someone might try to use `hostEndpoints`)

Comment (spikecurtis) [reply reference/resources/kubecontrollersconfig.md]: Thanks; fixed.

**Comment by spikecurtis** [comment]
@lmm this is ready for final review.  In particular, we need to merge this before any kube-controllers changes so that the CRD is available in KDD clusters for our tests.

**Pull request merged**